extensions/resume.md

```diff
@@ -0,0 +1,137 @@
+
+#### `RESUMED` Message
+
+This message is sent when a client successfully resumes a connection. For clients that have not completed connection registration, this indicates that their `RESUME` attempt was successful. For registered clients, this indicates that the given client disconnected and reconnected to the server.
+
+This message has the following format:
+
+    RESUMED nick user host timestamp
+
+`nick` is the nickname of the client whose connection has been resumed. `user` and `host` are the username and hostname that the new connecting client has. The client receiving this message MUST update the username and hostname they have stored for the given client, in the same way that the `CHGHOST` message is parsed.
+
+### Numerics
+
+Here are the numerics that this extension defines:
+
+| No. | Label                   | Format                                                          |
+| --- | ----------------------- | --------------------------------------------------------------- |
+| ??? | `ERR_CANNOT_RESUME`     | `:<server> ??? <oldnick> :Cannot resume connection, <details>`  |
+| ??? | `ERR_HISTORY_TRUNCATED` | `:<server> ??? <client/channel> :Chat history may be truncated` |
+
+`ERR_CANNOT_RESUME` is used to indicate a general failure to resume a connection for a specific reason. The purpose of this numeric is to pass along human-readable information informing the user why their connection wasn't automatically resumed such as a lack of SASL authentication and login. If a more appropriate numeric exists for conveying the issue already exists (such as `ERR_NOSUCHNICK` to represent the old nickname no longer being present on the network), then it should be used instead.
+
+Implementations should keep in mind that the only concrete indication that a `RESUME` attempt failed is that it does not receive a `RESUMED` message during registration. Clients MUST NOT rely on `ERR_CANNOT_RESUME` or any other numeric being received to trigger their 'resume has failed' state and process.
+
+`ERR_HISTORY_TRUNCATED` is used to indicate that chat history may be truncated for a given client/channel. It may be sent during a `chathistory` batch or otherwise during history playback. Bouncers that reconnect may also find this useful to send to their connected clients, to indicate that there may be missed history.
+
+
+## Connection Registration
+
+Upon connection, clients request the `draft/resume` capability. If they receive both this capability and the SASL capability, they can continue attempting to resume their old connection.
+
+After sending the `NICK` and `USER` commands, clients wishing to resume an old connection send an appropriate `RESUME` command indicating the nickname they wish to take over and how long they've been disconnected for.
```

Comment:
Does requesting `resume` hold registration until CAP END, or does it rely on the registration being held by `sasl`?